src/services/RESTService.ts

```diff
@@ -191,18 +111,32 @@ export class RESTService extends AbstractService {
     ]}
   }
 
+  /**
+   * It creates the RepositoryFactory used to build the http repository/clients and listeners.
+   * @param url the url.
+   */
+  public static createRepositoryFactory(url: string): RepositoryFactory {
+    //TODO Extend RepositoryFactoryHttp in tha TS SDK to allow ws:// kind of urls.
+    //TODO Why the WebSocket as webSocketInjected? Can we avoid this?
```

Comment:
no, this is necessary when running in the browser

Reply:
@fboucquez client builds use the browser's `WebSocket` class. Would require some testing to remove this but since the wallet is *always* browser served, I believe it could be removed (please mind the testing with _artifacts_ too)

Reply:
The TODOs are more for the SDK than for the wallet.  I'll find the way to configure the RepositoryFactory so you can tell which WebSocket you want to use and if the URL is HTTP: or ws: